sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -595,8 +595,58 @@ class Analyzer(
       }
     }
 
+    private def tryResolveHavingCondition(
+        a: UnresolvedHaving, havingCond: Expression, agg: LogicalPlan): LogicalPlan = {
+      val aggForResolving = agg match {
+        case a: Aggregate =>
+          // For CUBE/ROLLUP expressions, to avoid resolving repeatedly, here we delete them from
+          // groupingExpressions for condition resolving.
+          a.copy(groupingExpressions = Seq.empty)
```

Comment:
this looks hacky, can we get the original group by exprs from `Cube`/`RollUp`?

Reply:
Thanks, done in f2a2ad1.